source-container-build/app/test_source_build.py

```diff
@@ -783,6 +796,29 @@ def test_gather_go_deps(self):
         ]
         self._test_gather_deps_by_package_manager(gomod_deps, [".zip"])
 
+    def test_gather_srpm_deps(self):
+        srpm_deps = [
+            "output/bom.json",
+            "output/.build-config.json",
+            "output/x86_64/fedora/gpm-libs-1.20.7-42.fc38.x86_64.rpm",
+            "output/x86_64/fedora/repodata/bafa928710b0d4330ecb0d3d1936b17c2839d6aa-other.xml.zst",
+            "output/x86_64/fedora/repodata/repomd.xml",
+            "output/x86_64/fedora/repodata/00a7cbdb8c149280b63bb2e2abeaff85ab87ea-primary.xml.zst",
+            "output/x86_64/updates/vim-enhanced-9.1.113-1.fc38.x86_64.rpm",
+            "output/x86_64/updates/repodata/feb684972c3ddd3f2e195b7ca692fdf7de4e0-primary.xml.zst",
+            "output/x86_64/updates/repodata/6339e0d504567dc14512f01324d07bbb123-filelists.xml.zst",
+            "output/x86_64/updates-source/vim-9.1.113-1.fc38.src.rpm",
+            "output/x86_64/fedora-source/gpm.spec",
+            "output/x86_64/fedora-source/gpm-1.20.7-rhbz-668480-gpm-types-7-manpage-fixes.patch",
+            "output/x86_64/fedora-source/gpm.service",
+            "output/x86_64/fedora-source/gpm-1.20.7-42.fc38.src.rpm",
+            "output/x86_64/fedora-source/gpm-1.20.1-lib-silent.patch",
+            "output/x86_64/fedora-source/gpm-1.20.7.tar.xz",
+            "output/x86_64/fedora-source/1.20.7...e82d1a653ca94aa4ed12441424da6ce780b1e530.diff",
+            "output/x86_64/fedora-source/gpm-1.20.1-weak-wgetch.patch",
```

Comment:
Does rpm prefetch implementation extract the content of SRPMs?

Reply:
No, it does not. It just downloads them and stores them in the storage.
From rpm files, headers are read afterwards to get enough information for generating SBOM - and it is not even related to SRPMs, just RPMs.

Reply:
I realized I made a mistake in the demonstration version of rpm prefetcher.
Directories containing sources should be together in "sources [1]":

```
output/<arch1>/<repoid>/Packages/*.rpm
output/<arch1>/sources/*.src.rpm

```
[1] https://github.com/containerbuildsystem/cachi2/discussions/465